core/app/models/spree/promotion/rules/item_quantity.rb

```diff
@@ -0,0 +1,51 @@
+# A rule to apply to an order greater than (or greater than or equal to)
+# a specific quantity of line items
+module Spree
+  class Promotion
+    module Rules
+      class ItemQuantity < PromotionRule
+        preference :quantity_min, :integer, default: 1
+        preference :operator_min, :string, default: '>'
+        preference :quantity_max, :integer, default: 1000
+        preference :operator_max, :string, default: '<'
+
+        OPERATORS_MIN = ['gt', 'gte']
+        OPERATORS_MAX = ['lt', 'lte']
```

Comment:
Is this necessary? You can't have `.5` of a line item and this adds a fair bit of logic.